test/selenium_helpers.js

```diff
@@ -34,25 +31,163 @@ const setupSelenium = function(done) {
       })
     }
 
-  this.getLogs = () => {
-    return this.browser.manage().logs().get('browser')
+  browser.getLogs = function(){
+    return this.manage().logs().get('browser')
       .then(logs => console.log('Browser Logs:', logs))
   }
 
-  this.loginAs = loginAs
-  done()
+  browser.loginAs = function(githubId) {
+    return this.visit(`/__login/${githubId}`)
+    .then(session => {
+      const body = this.findElement(By.css('body'), 2000)
+      return this.wait(
+        until.elementTextContains(body, `logged in as ${githubId}`),
+        5000
+      )
+    })
+  }
+
+  browser.closeTabs = function(){
+    return this.getAllWindowHandles()
+      .then(handles => {
+        for(let i = handles.length -1; i > 0; i--){
+          this.switchTo().window(handles[i])
+          this.close()
+        }
+        this.switchTo().window(handles[0])
+      })
+  }
+
+  browser.checkNumberOfTabs = function() {
+    return this.getAllWindowHandles()
+      .then(handles => handles.length < 2)
+  }
+
+  browser.identifyTabPopup = function(timeout=2000) {
+    if(timeout <= 0) throw new Error('Out of time')
+    const _this = this
+    return this.checkNumberOfTabs()
+      .then(condition => {
+          if (condition) return Promise.resolve()
+          else{
+            setTimeout(function(){_this.identifyTabPopup(timeout-100)}, 100)
+          }
+      })
+  }
```

Comment:
However the recursion within promises is broken here. You cannot use setTimeout because it breaks the promise chain. I think you need to do something like this:

```js
browser.identifyTabPopup = function(timeout=2000) {
  if (timeout <= 0) throw new Error('Out of time')
  return this.checkNumberOfTabs()
    .then(condition => {
      if (condition) return Promise.resolve()
      return this.sleep(100)
        .then(() => this.identifyTabPopup(timeout - 100))
    })
}
```